torch/csrc/jit/import.cpp

```diff
@@ -153,7 +153,14 @@ at::Tensor ScriptModuleDeserializer::loadTensor(const torch::TensorDef& tensor_p
           at::DeviceTypeName(device.type(), false));
     }
   }
-  AT_ASSERT(storage_it->second.device() == device);
+  if (storage_it->second.device().type() != device.type() ||
+      (device.has_index() &&
+       storage_it->second.device().index() != device.index())) {
+    std::stringstream oss;
```

Comment:
nits: AT_ERROR is vararg so there is no need for a stringstream. The error message is also not very informative: "storage previously was specified with device storage_it->second.device() but now is specified with device device"

Reply:
Have updated the error message. I didn't find a good way to convert at::Device to string, except the stringstream.